TESTING.md

```diff
@@ -11,4 +13,102 @@ START /B /W generalszh.exe -jobs 4 -headless -replay subfolder/*.rep > replay_ch
 echo %errorlevel%
 PAUSE
 ```
-It will run the game in the background and check that each replay is compatible. You need to use a VC6 build with optimizations and RTS_BUILD_OPTION_DEBUG = OFF, otherwise the game won't be compatible.
\ No newline at end of file
+It will run the game in the background and check that each replay is compatible. You need to use a VC6 build with optimizations and RTS_BUILD_OPTION_DEBUG = OFF, otherwise the game won't be compatible.
+
+## Code Quality Analysis with clang-tidy
+
+The project includes clang-tidy configuration for static code analysis to help maintain code quality and catch potential bugs.
+
+### Prerequisites
+
+1. **CMake with compile commands export**: The CMake presets already have `CMAKE_EXPORT_COMPILE_COMMANDS=ON` configured.
+2. **clang-tidy**: Install clang-tidy for your platform:
+   - **Linux**: `apt install clang-tidy` or `yum install clang-tools-extra`
+   - **Windows**: Install LLVM or use the version that comes with Visual Studio
+   - **macOS**: `brew install llvm` or use Xcode command line tools
+
+### Running clang-tidy
+
+#### Method 1: Using the helper script (Recommended)
+
+The project includes a Python script that simplifies running clang-tidy:
+
+```bash
+# Analyze all source files
+python3 scripts/run-clang-tidy.py
+
+# Analyze only Core directory
+python3 scripts/run-clang-tidy.py --include Core/
+
+# Analyze GeneralsMD but exclude certain patterns
+python3 scripts/run-clang-tidy.py --include GeneralsMD/ --exclude test
+
+# Use specific build directory
+python3 scripts/run-clang-tidy.py --build-dir build/win32
+
+# Apply fixes automatically (use with caution!)
+python3 scripts/run-clang-tidy.py --fix --include Core/Libraries/
+```
+
+#### Method 2: Direct clang-tidy usage
+
+First, ensure you have a `compile_commands.json` file:
+
+```bash
+# Configure with any preset that exports compile commands
+cmake --preset win32  # or vc6, unix, etc.
+
+# Run clang-tidy on specific files
+clang-tidy -p build/win32 Core/Libraries/Source/RTS/File.cpp
+
+# Run on multiple files with pattern
+find Core/ -name "*.cpp" | xargs clang-tidy -p build/win32
```

Comment:
Does this command work in Windows Command Prompt?

Reply:
New commit added to clarify the Windows commands.